### PR TITLE
Fix #19: rate limit issue

### LINKED ISSUE #19: Throttle::attempt() goes over limit before returning "false"
With the following code:

```php
if (!Throttle::attempt($request, 5)) {
    throw new Exception('Over limit');
}

doSomething();
```

The exception is not thrown on the sixth attempt as expected, but on the seventh. The issue seems related to issue #6.

### PR DESCRIPTION
**Pull request by lazychaser**
`attempt` performs `check` first and then `hit`
`check` condition changed

### CODE REVIEW

File: src/ThrottleServiceProvider.php [hunk 1]
```diff
@@ -52,15 +52,14 @@ protected function setupConfig()
...
      *
      * @return void
      */
-    protected function setupFilters(Router $router, Throttle $throttle)
+    protected function setupFilters(Router $router)
     {
-        $router->filter('throttle', function ($route, $request, $limit = 10, $time = 60) use ($throttle) {
-            if (!$throttle->attempt($request, $limit, $time)) {
+        $router->filter('throttle', function ($route, $request, $limit = 10, $time = 60) {
+            if (! $this->app->throttle->attempt($request, $limit, $time)) {
```
Comment (GrahamCampbell) [inline src/ThrottleServiceProvider.php]: no space after the ! please

**Comment by GrahamCampbell** [comment]
Thank you so much for this. :)

**Comment by GrahamCampbell** [comment]
I'll be releasing 4.0.0 soonish to include this.

**Comment by lazychaser** [comment]
Good, thanks!
24 Апр 2015 г. 22:16 пользователь "Graham Campbell" <
notifications@github.com> написал:

**Comment by GrahamCampbell** [comment]
In the meanwhile, you can use `~4.0@dev` or something similar to load this package. :)

**Pull request merged**